lib/services/platform-project-service-base.ts

```diff
@@ -0,0 +1,25 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import path = require("path");
+
+export class PlatformProjectServiceBase implements IPlatformProjectServiceBase {
+	constructor(protected $fs: IFileSystem) { }
+	
+	public getPluginPlatformsFolderPath(pluginData: IPluginData, platform: string) {
+		return pluginData.pluginPlatformsFolderPath(platform);
+	}
+	
+	public getAllNativeLibrariesForPlugin(pluginData: IPluginData, platform: string, filter: (fileName: string, pluginPlatformsFolderPath: string) => boolean): IFuture<string[]> {
+		return (() => {
+			let pluginPlatformsFolderPath = this.getPluginPlatformsFolderPath(pluginData, platform);
+			if(pluginPlatformsFolderPath && this.$fs.exists(pluginPlatformsFolderPath).wait()) {
+				let platformsContents = this.$fs.readDirectory(pluginPlatformsFolderPath).wait();
+				return _(platformsContents)
```

Comment:
this logic can be rewritten with a local `string[]` variable  which is assigned to and then returned as to avoid a return statement inside the if block and to have only one return statement at the end of the function